command_server.sh

```
@@ -122,6 +122,22 @@ $1
 	echo "{\"Done\": true, \"Exit\": $__cmd_last_status, \"Dir\": \"$dir\"}"
 }
 
+__cmd_complete() {
+    prog="${1%% *}"
+    if [[ "${#1}" -gt "${#prog}" ]]; then
+        # argument parsing
+        comp_command="$(complete -p "$prog" | sed s/^complete/compgen/)"
```

> `$(complete -p "$prog" | sed s/^complete/compgen/)`

~~So I'm pretty sure this will spawn a subshell, which will cause that injected tcsetpgrp hook to try to print out process group changes as a new bash process is launched. I guess it just fails silently, so maybe that's not a big deal? You might consider trying to do this without a subshell. Totally understand if you want to put this off. :)~~

Edit: Actually, I'm unsure whether tcsetpgrp actually gets called in this scenario, since the command substitution isn't a real job. Worth experimenting with at some point in the future.

You should do string replacement in bash without shelling out to sed, though.